### PR TITLE
Feature/search page

### PR DESCRIPTION
**Pull request by AdrielGS**
### Verify that
Git checklist
- [ ] the branch name follows the git flow naming convention + label + UT reference if applicable (feature/i18n/UT-283)
- [ ] I have rebased my branch to develop before submitting my PR

Front end checklist
- [ ] the design is responsive
- [ ] the code doesn't break Server Side Rendering

API checklist
- [] the code doesn't break the api endpoints contract (applicable when we're in production and the api is officially open and used by 3rd party app)

Tests and comments checklist
- [X ] JS Standard code style has been verified with ESLint
- [ ] the pull request contains unit tests for new code that passes and has at least 50% coverage
- [ ] functions are prefixed with JSDoc information or the doc has been updated

### My PR doesn't concern a Jira task
Create a new article search page 

---

## Test plan
- Write in the top bar an article you want to search, for instance: "test".
- The page will load with some results (if you are using the test database).
- After loading you can add some filters such as Category, Tags, and Language.

_The language filter is already working with the new field from the Article Collection. So it will only work if the article has the "lang" field with an value of "eng", for instance._

### CODE REVIEW

File: packages/api/modules/search/handlers.js [hunk 1]
```diff
@@ -11,7 +12,34 @@ const Article = require('../articles/article.model')
...
+
+  if (req.payload.project && req.payload.project.length > 0) {
+    optionalConditions.project = req.payload.project
+  }
+
+  const articles = await Article.find({
+    title: { '$regex': req.payload.title, '$options': 'i' },
+    ...optionalConditions })
+    .sort({ createdAt: req.payload.sortBy })
+    .limit(req.payload.limit)
```
Comment (gregory-latinier) [inline packages/api/modules/search/handlers.js]: Where is the `skip` instruction? You need to return only the result of the next page 
In this situation if I want the page 1000 you'll just return 20 000 results... this is not pagination

File: packages/client/src/pages/search/articles/search-articles.vue [hunk 2]
```diff
@@ -15,15 +170,39 @@ export default {
...
+          q-chips-input(
+            v-model="searchForm.tags"
+            @duplicate="duplicatedTags"
+            @input="chipsInputChange"
+            :placeholder="searchForm.tags.length === 0 ? 'Filter by tags' : ''"
+          )
+            q-autocomplete(@search="tagsAutocomplete", :min-characters="2", :max-results="10")
+        q-field(orientation="vertical", :label="$t('search.searchForm.languages.label')", :count="5")
+          q-select(multiple, filter, chips, v-model="searchForm.languages", :options="languages", :placeholder="$t('search.searchForm.languages.placeholder')", @input="languagesInputChange")
+        div.flex.justify-center
```
Comment (gregory-latinier) [inline packages/client/src/pages/search/articles/search-articles.vue]: useless div
`div.flex` is the same as `.flex`

File: packages/client/src/pages/search/articles/search-articles.vue [hunk 3]
```diff
@@ -15,15 +170,39 @@ export default {
...
+          q-btn(color="primary", :label="$t('search.clearAll.label')", @click="clearFilters")
     .col-md-9
-      | Search results
-    
+      div.flex.justify-end
```
Comment (gregory-latinier) [inline packages/client/src/pages/search/articles/search-articles.vue]: same

File: packages/client/src/pages/search/articles/search-articles.vue [hunk 4]
```diff
@@ -15,15 +170,39 @@ export default {
...
+        q-select.sort-by(v-model="searchForm.sortBy" , :options="sortByOptions", placeholder="$t('search.sortBy.placeholder')", @input="() => {resetFlags();filterArticles()}")
+      u-article-card(v-for="article in articles", :key="article._id", :article="article")
+      div.flex.justify-center
```
Comment (gregory-latinier) [inline packages/client/src/pages/search/articles/search-articles.vue]: same

File: packages/client/src/store/search/actions.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ export const searchArticles = async (context, search) => {
     context,
     method: 'post',
     url: '/v1/search/articles',
-    data: { search }
+    data: search
```
Comment (gregory-latinier) [inline packages/client/src/store/search/actions.js]: data for shorthand

File: packages/client/src/pages/search/articles/search-articles.vue [hunk 6]
```diff
@@ -1,12 +1,167 @@
 <script>
-import { mapGetters } from 'vuex'
+import { mapGetters, mapActions } from 'vuex'
+import { LanguagesMixin } from 'src/mixins/languages'
+import { maxLength, required } from 'vuelidate/lib/validators'
+import UFormProject from 'src/components/form/project'
```
[the remark this answers is not in the document]
Comment (AdrielGS) [reply packages/client/src/pages/search/articles/search-articles.vue]: Just a reminder that these components' name, and all of the others external components, will be changed in another PR

**Pull request merged**